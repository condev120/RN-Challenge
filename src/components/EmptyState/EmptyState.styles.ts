import { StyleSheet } from "react-native";
import { colors, fontWeights, sizes } from "../../constants/styles";

export const emptyStateStyles = StyleSheet.create({
  image: {
    height: 150,
    width: 200,
    marginBottom: 10
  },
  title: {
    fontSize: sizes.xl,
    color: colors.black,
    fontWeight: fontWeights.bold,
    marginBottom: 0
  },
  message: {
    marginTop: 0,
    fontSize: sizes.md,
    color: colors.neutral,
    marginBottom: 14
  },
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
    gap: 10
  },
  button: {
    backgroundColor: colors.black,
    paddingHorizontal: 30,
    paddingVertical: 16,
    borderRadius: 16
  },
  buttonText: {
    color: colors.white,
    fontSize: sizes.md,
    fontWeight: fontWeights.bold
  }
})
