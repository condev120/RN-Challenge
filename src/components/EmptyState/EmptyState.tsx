import {
  Image,
  ImageSourcePropType,
  Text,
  TouchableOpacity,
  View
} from 'react-native'
import { FC } from 'react'
import { emptyStateStyles } from './EmptyState.styles'

export interface IEmptyStateProps {
  image: ImageSourcePropType
  title: string
  message: string
  actionLabel: string
  onAction: () => void
}

export const EmptyState: FC<IEmptyStateProps> = ({
  image,
  title,
  message,
  actionLabel,
  onAction
}) => {
  return (
    <View style={emptyStateStyles.container}>
      <Image source={image} style={emptyStateStyles.image} />
      <Text style={emptyStateStyles.title}>{title}</Text>
      <Text style={emptyStateStyles.message}>{message}</Text>
      <TouchableOpacity
        style={emptyStateStyles.button}
        activeOpacity={0.7}
        onPress={() => onAction()}
      >
        <Text style={emptyStateStyles.buttonText}>{actionLabel}</Text>
      </TouchableOpacity>
    </View>
  )
}
