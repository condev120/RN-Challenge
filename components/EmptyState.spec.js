import React from 'react';
import { render, act, waitFor, fireEvent } from '@testing-library/react-native';
import EmptyState from './EmptyState';

let rendered;
let mockAction = jest.fn();
describe('EmptyState - Coverage', () => {
  beforeEach(async () => {
    await waitFor(async () => {
      rendered = render(<EmptyState image={require('../assets/empty-discover.jpg')}
      title="No results found"
      message="Try adjusting the settings"
      actionLabel="Go to Settings"
      onAction={() => mockAction()}/>);
    });
  });

  it('Renders correctly', async () => {
    expect(rendered.toJSON()).toMatchSnapshot();
    expect(rendered.getByText('No results found')).toBeDefined();
    expect(rendered.getByText('Try adjusting the settings')).toBeDefined();
    expect(rendered.getByText('Go to Settings')).toBeDefined();
  });

  it('Button should be triggered correctly', () => {
    const button = rendered.getByText('Go to Settings');
    fireEvent.press(button);
    expect(mockAction).toBeCalledTimes(1);
  });
});
