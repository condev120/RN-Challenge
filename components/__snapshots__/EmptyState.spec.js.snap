// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState - Coverage Renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "gap": 10,
      "justifyContent": "center",
    }
  }
  testID="EmptyState"
>
  <Image
    source={
      {
        "testUri": "../../../assets/empty-discover.jpg",
      }
    }
    style={
      {
        "height": 150,
        "marginBottom": 10,
        "width": 200,
      }
    }
  />
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "700",
        "marginBottom": 0,
      }
    }
  >
    No results found
  </Text>
  <Text
    style={
      {
        "color": "#757575",
        "fontSize": 14,
        "marginBottom": 14,
        "marginTop": 0,
      }
    }
  >
    Try adjusting the settings
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#000000",
        "borderRadius": 16,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 14,
          "fontWeight": "700",
        }
      }
    >
      Go to Settings
    </Text>
  </View>
</View>
`;
